.clang-format

```diff
@@ -73,7 +74,7 @@ AlwaysBreakTemplateDeclarations: true
 BinPackArguments: false
 BinPackParameters: false
 BreakBeforeBinaryOperators: NonAssignment
-BreakBeforeBraces: Attach
+BreakBeforeBraces: Allman
```

Comment:
I have some thoughts on this :smiley: 

* My personal preference is `BreakBeforeBraces: Stroustrup`.
    * We can both agree that breaking before functions makes a lot of sense
    * I like the look and feel of `if (condition) {` all on one line
    * With Allman I find the else conditions are very awkward:

        ```c
          }
          else
          {
        ```

    * These feel much more natural with Stroustrup:

        ```c
          }
          else {
        ```

* FWIW, [Zephyr is on Linux style](https://github.com/zephyrproject-rtos/zephyr/blob/18c23dec5f60836fc84a9cc93293769dff84e5c2/.clang-format#L30)

Reply:
> I like the look and feel of `if (condition) {` all on one line

Do you find that it helps you to read the code better? I find that having the open and closing braces in the same column makes it easier to visually line them up. And having a mostly empty line at the start and end of a block helps delineate them. For example in this code, it's easier to see at a glance where the conditional stops and the body starts in Allman (2nd) vs Stroustrup (1st):

```
if (is_this_function_true() &&
    get_some_value() == MY_CONSTANT &&
    check_value(foo)) {
    call_a_function(bar);
    baz = x + 5;
    printf("something something something");
}
```

```
if (is_this_function_true() &&
    get_some_value() == MY_CONSTANT &&
    check_value(foo))
{
    call_a_function(bar);
    baz = x + 5;
    printf("something something something");
}
```

> With Allman I find the else conditions are very awkward

I agree they're verbose, but they don't interfere with my ability to read and understand the code.

Reply:
Yeah, that makes sense. I'm fine with going with the one we think is more readable.